.github/workflows/update-documentation.yml

```diff
@@ -0,0 +1,138 @@
+        id: gather-information
+        shell: bash
+        run: |
+          NUM_CHANGED_CHARTS=0
+          cd charts || exit 1
+          for chart in */; do
+            # Check if values.yaml has changed
+            if git diff --name-only HEAD~1 | grep -q "$chart"; then
+              echo "$chart: No changes detected"
+              continue
+            fi
+            echo "$chart: Changes detected"
+            NUM_CHANGED_CHARTS=$((NUM_CHANGED_CHARTS + 1))
+            echo "CHANGED_CHART=$chart" >> "$GITHUB_ENV"
+          done
+          if [ "$NUM_CHANGED_CHARTS" -eq 0 ]; then
+            echo "ERROR=No charts have changed" >> "$GITHUB_ENV"
+            echo "FAILED=true" >> "$GITHUB_ENV
+          fi
+          if [ "$NUM_CHANGED_CHARTS" -gt 1 ]; then
+              echo "ERROR=Multiple charts have changed, only update one Chart per PR" >> "$GITHUB_ENV"
+              echo "FAILED=true" >> "$GITHUB_ENV
+          fi
+          if git diff --name-only HEAD~1 | grep -q "$CHANGED_CHART/values.yaml"; then
+            echo "$CHANGED_CHART: No values.yaml changes detected"
+          fi
+          echo "$CHANGED_CHART: values.yaml changes detected"
+          echo "VALUES_CHANGED=true" >> "$GITHUB_ENV"
+      # Using actions/github-scripts because using exit 1 in the script above would not provide any output
+      # Source: https://github.community/t/no-output-on-process-completed-with-exit-code-1/123821/3
+      - id: show-error
+        name: Show error
+        if: ${{ steps.gather-information.outputs.failed }}
+        uses: actions/github-script@7.0.1
+        with:
+          script: |
+            if (!${{ steps.gather-information.outputs.error }}) {
+              core.setFailed(${{ steps.gather-information.outputs.error }});
+            }
+  update-documentation:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs: gather-information
+    if: ${{ needs.gather-information.outputs.failed != 'true' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          token: ${{ secrets.CONTANE_BOT_TOKEN }}
+      - name: Configure Git
+        run: |
+          git config user.name "contane-bot"
+          git config user.email "160241315+contane-bot@users.noreply.github.com"
+      - name: Install readme-generator-for-helm
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        run: npm install -g @bitnami/readme-generator-for-helm@2.6.1
+      - name: Update README.md
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        shell: bash
+        run: |
+          cd charts || exit 1
+          echo "Updating README.md for $CHANGED_CHART"
+          readme-generator --values "$CHANGED_CHART/values.yaml" --readme "$CHANGED_CHART/README.md"
```

Comment:
This seems potentially dangerous unless we can be 100% sure of the validity of `$CHANGED_CHART`, since a user input is part of a file path, which could result in a directory traversal.